FireboltNETSDK/FireboltClient.cs

```diff
@@ -285,37 +293,82 @@ protected async Task<T> SendAsync<T>(HttpMethod method, string uri, HttpContent?
         }
     }
 
-    private void ReconnectIfNeeded(HttpResponseHeaders headers)
+    private void ProcessResponseHeaders(HttpResponseHeaders headers, CancellationToken cancellationToken)
```

Comment:
I would split this method into processing headers step and updating parameters step, each its own method. This would ensure better modularity and make this method easier to comprehend.

Reply:
Done. I was thinking about this but was too lazy to implement. I have given me the required trigger. :)